### DISCUSSION TITLE
Impact of the "Activate background import" switch when creating a new customer in the back office

### DISCUSSION THREAD

**Discussion by cwm77** [monarc-project/MonarcAppFO#525] [Q&A] [answered]
Hello,

I am hoping for an answer to the question as to what effect setting the "Activate background import" switch in the back office has when a new customer is created. Please find attached a screenshot which should clarify where exactly this switch is located in the back office. 

Many thanks and best regards

![image](https://github.com/monarc-project/MonarcAppFO/assets/9320205/90057cba-f83f-471e-92ff-24b17a9f0172)

**Comment by ruslanbaidan** [comment, marked as answer]
Hello,
The complete analysis import can take a long time and fails due to some proxy waiting or process allocated time limitations.
The purpose of the background import is to avoid the server waiting time when a new import is started by initiating a process that will be launched in a background mode.
When the configuration option is one there and a new import is started, there will be a redirect to the list of analyses with a displayed loader. The loader signals that the process is ongoing and the analysis is unaccessible until it's finished.
The option is available on clients modifications as well.

To configure the background import you need to:
- Use the latest [ansible code changes](https://github.com/monarc-project/ansible-ubuntu/blob/master/playbook/monarcfo/templates/client_config.php.j2#L108-L109), where the option is applied in the clients' configs.
- Create a cronjobs on the clients side described in the updating section [here](https://www.monarc.lu/news/2023/10/25/monarc-2127-released/).
